### ISSUE TITLE
Server should translate xpath selectors without namespaces to selectors with namespaces

### ISSUE THREAD

**Issue by GoogleCodeExporter** [steven-h-gardiner/gae-wrapup-server#2]
```

The situation with namespaces is confusing.

Briefly, in a document an element tagname without a prefix is interpreted as 
belonging to the default namespace at that point in the document.  In an xpath, 
an element tagname without a prefix is interpreted (by 
[https://github.com/ilinsky/jquery-xpath the xpath jquery plugin], and possibly 
by the browser?)as belonging to no namespace. 

So //table//tr/td[2] does not match the second column of an HTML table as 
expected if the HTML table uses the XHTML namespace.

XPath does not provide a mechanism for referencing the default namespace; it 
only provides mechanisms for using a specific namespace, or for matching any 
namespace with "*"

To workaround this, I use the convention that the prefix "_" always refers to 
the document's default namespace, or no namespace if the document does not have 
one.  

So basically all wrappers stored in the server that use xpath selectors should 
always use //_:table//_:tr/_:td[2] instead of the xpath above.

It's not hard to translate between the two forms, but it does feel like 
something software should do for you.  I think the server should rewrite 
automatically, either when the wrapper is written to the repository, or when it 
is retrieved.

Note that a regular expression based approach seems easy (e.g. put '_:' after 
'/') but there are several edge cases so it's probably easier to find some code 
that parses an XPath and modify the parse tree.

Examples of harder XPaths
//table//tr/td[preceding-sibling::td/@width='40%']

```

Original issue reported on code.google.com by `steven.h...@gmail.com` on 20 Aug 2014 at 7:30
